test/unit/metrics/test_metricframe_bootstrap.py

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+# These functions are borrowed from "test_metricframe_by_group.py"
+# with additional checks for bootstrap related operations.
+
+import numpy as np
+import pandas as pd
+import pytest
+import sklearn.metrics as skm
+
+from fairlearn.metrics import MetricFrame
+
+from .data_for_test import y_t, y_p, s_w, g_1, g_2, g_3, g_4
+
+
+metric = [
+    skm.recall_score,
+    skm.precision_score,
+    skm.accuracy_score,
+    skm.balanced_accuracy_score,
+]
+
+ci_options = [0.95, 0.05, [0.2, 0.4, 0.6, 0.8]]
+ci_methods = ["percentile", "bias-corrected"]
+
+# Bootstrap test helper functions
+
+
+def check_all_bootstrap_output_shapes(mf):
+    # Check if all .*_ci outputs are same shape as regular outputs
+    assert all(mf.overall.shape == ci_output.shape for _, ci_output in mf.overall_ci)
+    assert all(mf.by_group.shape == ci_output.shape for _, ci_output in mf.by_group_ci)
+    assert all(
+        mf.group_min().shape == ci_output.shape for _, ci_output in mf.group_min_ci
+    )
+    assert all(
+        mf.group_max().shape == ci_output.shape for _, ci_output in mf.group_max_ci
+    )
+    assert all(
+        mf.difference(method="to_overall").shape == ci_output.shape
+        for _, ci_output in mf.difference_overall_ci
+    )
+    assert all(
+        mf.difference(method="between_groups").shape == ci_output.shape
+        for _, ci_output in mf.difference_group_ci
+    )
+    assert all(
+        mf.ratio(method="to_overall").shape == ci_output.shape
+        for _, ci_output in mf.ratio_overall_ci
+    )
+    assert all(
+        mf.ratio(method="between_groups").shape == ci_output.shape
+        for _, ci_output in mf.ratio_group_ci
+    )
+
+
+def check_all_bootstrap_directions(mf):
```

Comment:
This should probably check that the values reported by `overall`, `difference()` etc. lie within the confidence intervals